src/OpenApi/src/Services/Schemas/OpenApiSchemaService.cs

```diff
@@ -82,9 +82,9 @@ internal sealed class OpenApiSchemaService(
                 };
             }
             // STJ uses `true` in place of an empty object to represent a schema that matches
-            // anything. We override this default behavior here to match the style traditionally
-            // expected in OpenAPI documents.
-            if (type == typeof(object))
+            // anything (like the `object` type) or types with user-defined converters. We override
+            // this default behavior here to match the style traditionally expected in OpenAPI documents.
```

Comment:
What does "traditionally" mean here?  Is there a newer style of OpenAPI documents?

Reply:
Yes, OpenAPI v3.1 is an upcoming version that has full parity with JSON schema and would permit `true` schemas being used as catch-all schemas. I'll update this comment to be more explicit.